packages/nest/src/oauth/data/datasource/query/oauth-access-token.query.ts

```diff
@@ -7,7 +7,7 @@ export class OAuthAccessTokenQuery extends SQLWhereQuery {
     whereParams(): any[] {
         return [this.accessToken];
     }
-    whereSql(dialect: SQLDialect): string {
-        return `access_token = ${dialect.getParameterSymbol(1)}`;
+    whereSql(dialect: SQLDialect, params: SQLQueryParamComposer): string {
+        return `access_token = ${params.next()}`;
```

Comment:
I'm thinking that if the `SQLDialect` was needed for `getParameterSymbol`, then this signature should be simplified and just pass `SQLQueryParamComposer` as that's the one taking care of this now.

This would be a breaking change, we could pack it with the `next(value)` change.

Reply:
This is an awesome idea @doup. We could remove the dupclication of logic if adding parameters in a method, and adding the SQL syntax in another. 

However, we must think smart and do it in a retro-compatible way, otherwise it might break projects. Today I have to add a couple of changes to harmony-nest, and I'll think about solutions.

Reply:
Resolving in favor of: https://github.com/mobilejazz/harmony-typescript/issues/48